src/commands/ProposalCommand.ts

```diff
@@ -0,0 +1,79 @@
+import { Command } from "../interactions/interactionClasses";
+import {
+	ChatInputCommandInteraction,
+	SlashCommandBuilder,
+	SlashCommandStringOption,
+	SlashCommandSubcommandsOnlyBuilder,
+} from "discord.js";
+import { logger } from "../logger";
+import { addProposal, deleteProposal, editProposal } from "../util/proposal/manageProposals";
+import { listProposals, viewProposal } from "../util/proposal/listProposals";
+
+class ProposalCommand extends Command {
+	constructor() {
+		super("proposal");
+	}
+
+	async execute(interaction: ChatInputCommandInteraction): Promise<void> {
+		if (interaction.channel?.isThread()) {
+			await interaction.reply({
+				content:
+					"These commands are not allowed in threads. Please use them in the main channel for everyone to see them.",
+				ephemeral: true,
+			});
+			return;
+		}
+
+		const subcommand = interaction.options.getSubcommand();
+
+		const commands: { [key: string]: (interaction: ChatInputCommandInteraction) => Promise<void> } = {
+			add: addProposal,
+			delete: deleteProposal,
+			edit: editProposal,
+			view: viewProposal,
+			list: listProposals,
+		};
+
+		if (commands[subcommand]) {
+			await commands[subcommand](interaction);
+		} else {
+			await interaction.reply("I don't know that subcommand... Please contact a developer.");
```

Comment:
Why do the check here? If a command doesn't exist you cant execute it in the first place right?

Reply:
The check is basically for the devs to notice when they messed up with the names of the subcommands. Since the names are working fine and all the commands where tested I can remove the if :)